
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <meta name="viewport" content="width=device-width,initial-scale=1">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <meta name="lang:clipboard.copy" content="Copy to clipboard">
  <meta name="lang:clipboard.copied" content="Copied to clipboard">
  <meta name="lang:search.language" content="en">
  <meta name="lang:search.pipeline.stopwords" content="True">
  <meta name="lang:search.pipeline.trimmer" content="True">
  <meta name="lang:search.result.none" content="No matching documents">
  <meta name="lang:search.result.one" content="1 matching document">
  <meta name="lang:search.result.other" content="# matching documents">
  <meta name="lang:search.tokenizer" content="[\s\-]+">

  
    <link href="https://fonts.gstatic.com/" rel="preconnect" crossorigin>
    <link href="https://fonts.googleapis.com/css?family=Roboto+Mono:400,500,700|Roboto:300,400,400i,700&display=fallback" rel="stylesheet">

    <style>
      body,
      input {
        font-family: "Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif
      }

      code,
      kbd,
      pre {
        font-family: "Roboto Mono", "Courier New", Courier, monospace
      }
    </style>
  

  <link rel="stylesheet" href="_static/stylesheets/application.css"/>
  <link rel="stylesheet" href="_static/stylesheets/application-palette.css"/>
  <link rel="stylesheet" href="_static/stylesheets/application-fixes.css"/>
  
  <link rel="stylesheet" href="_static/fonts/material-icons.css"/>
  
  <meta name="theme-color" content="#00020a">
  <script src="_static/javascripts/modernizr.js"></script>
  
  
  
    <title>7.1.8. Prepare MR Ensemble with CCP4mg &#8212; CCP4 Cloud 1.7 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/material.css" />
    <link rel="stylesheet" type="text/css" href="_static/contentui.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/contentui.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="7.2. Phaser" href="doc.task.Phaser.html" />
    <link rel="prev" title="7.1.7. Prepare MR Ensemble from Coordinate Data" href="doc.task.EnsemblePrepXYZ.html" />
  
   

  </head>
  <body dir=ltr
        data-md-color-primary=#005BBB data-md-color-accent=yellow>
  
  <svg class="md-svg">
    <defs data-children-count="0">
      
      <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
      
    </defs>
  </svg>
  
  <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer">
  <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search">
  <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
  <a href="#doc.task.EnsemblePrepMG" tabindex="1" class="md-skip"> Skip to content </a>
  <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex navheader">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="index.html" title="CCP4 Cloud 1.7 documentation"
           class="md-header-nav__button md-logo">
          
            <i class="md-icon">cloud</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          <span class="md-header-nav__topic">CCP4 Cloud 1.7 documentation</span>
          <span class="md-header-nav__topic"> 7.1.8. Prepare MR Ensemble with CCP4mg </span>
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
        
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" action="search.html" method="get" name="search">
      <input type="text" class="md-search__input" name="q" placeholder="Search"
             autocapitalize="off" autocomplete="off" spellcheck="false"
             data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>

      </div>
      
      
  
  <script src="_static/javascripts/version_dropdown.js"></script>
  <script>
    var json_loc = ""versions.json"",
        target_loc = "../",
        text = "Versions";
    $( document ).ready( add_version_dropdown(json_loc, target_loc, text));
  </script>
  

    </div>
  </nav>
</header>

  
  <div class="md-container">
    
    
    
  <nav class="md-tabs" data-md-component="tabs">
    <div class="md-tabs__inner md-grid">
      <ul class="md-tabs__list">
          <li class="md-tabs__item"><a href="index.html" class="md-tabs__link">CCP4 Cloud 1.7 documentation</a></li>
          <li class="md-tabs__item"><a href="Molecular_Replacement.html" class="md-tabs__link"><span class="section-number">7. </span>Molecular Replacement</a></li>
          <li class="md-tabs__item"><a href="doc.task.MR_model_preparation.html" class="md-tabs__link"><span class="section-number">7.1. </span>Molecular Replacement Models and Ensembles</a></li>
      </ul>
    </div>
  </nav>
    <main class="md-main">
      <div class="md-main__inner md-grid" data-md-component="container">
        
          <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
            <div class="md-sidebar__scrollwrap">
              <div class="md-sidebar__inner">
                <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="index.html" title="CCP4 Cloud 1.7 documentation" class="md-nav__button md-logo">
      
        <i class="md-icon">cloud</i>
      
    </a>
    <a href="index.html"
       title="CCP4 Cloud 1.7 documentation">CCP4 Cloud 1.7 documentation</a>
  </label>
  

  
  <ul class="md-nav__list">
    <li class="md-nav__item">
    
    
      <a href="Workflows.html" class="md-nav__link">1. Automatic workflows</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Data_Import.html" class="md-nav__link">2. Data Import</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Structure_prediction.html" class="md-nav__link">3. Structure Prediction</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Data_Processing.html" class="md-nav__link">4. Data Processing</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Asymmetric_Unit_and_Structure_Revision.html" class="md-nav__link">5. Asymmetric Unit and Structure Revision</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Automated_Molecular_Replacement.html" class="md-nav__link">6. Automated Molecular Replacement</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Molecular_Replacement.html" class="md-nav__link">7. Molecular Replacement</a>
      <ul class="md-nav__list"> 
    <li class="md-nav__item">
    
    
      <a href="doc.task.MR_model_preparation.html" class="md-nav__link">7.1. Molecular Replacement Models and Ensembles</a>
      <ul class="md-nav__list"> 
    <li class="md-nav__item">
    
    
      <a href="doc.task.MrParse.html" class="md-nav__link">7.1.1. Find and prepare MR models with MrParse</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.ModelPrepXYZ.html" class="md-nav__link">7.1.2. Prepare MR Model(s) from Coordinate data</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.ModelPrepAlgn.html" class="md-nav__link">7.1.3. Prepare MR Model(s) from Alignment data</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.ModelPrepMC.html" class="md-nav__link">7.1.4. Prepare Complex Model</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.Ensembler.html" class="md-nav__link">7.1.5. Prepare MR Ensemble from Models</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.EnsemblePrepSeq.html" class="md-nav__link">7.1.6. Prepare MR Ensemble from Sequence</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.EnsemblePrepXYZ.html" class="md-nav__link">7.1.7. Prepare MR Ensemble from Coordinate Data</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    <label class="md-nav__link md-nav__link--active" for="__toc"> 7.1.8. Prepare MR Ensemble with CCP4mg </label>
    
      <a href="#" class="md-nav__link md-nav__link--active">7.1.8. Prepare MR Ensemble with CCP4mg</a>
      
        
<nav class="md-nav md-nav--secondary">
    <label class="md-nav__title" for="__toc">Contents</label>
  <ul class="md-nav__list" data-md-scrollfix="">
        <li class="md-nav__item"><a href="#doc-task-ensembleprepmg--page-root" class="md-nav__link">7.1.8. Prepare MR Ensemble with CCP4mg</a><nav class="md-nav">
              <ul class="md-nav__list">
        <li class="md-nav__item"><a href="#parameters" class="md-nav__link">7.1.8.1. Parameters</a>
        </li>
        <li class="md-nav__item"><a href="#workflow" class="md-nav__link">7.1.8.2. Workflow</a>
        </li></ul>
            </nav>
        </li>
    
<li class="md-nav__item"><a class="md-nav__extra_link" href="_sources/doc.task.EnsemblePrepMG.rst.txt">Show Source</a> </li>

  </ul>
</nav>
      <ul class="md-nav__list"> 
    <li class="md-nav__item">
    
    
      <a href="#parameters" class="md-nav__link">7.1.8.1. Parameters</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="#workflow" class="md-nav__link">7.1.8.2. Workflow</a>
      
    
    </li></ul>
    
    </li></ul>
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.Phaser.html" class="md-nav__link">7.2. Phaser</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="doc.task.Molrep.html" class="md-nav__link">7.3. Molrep</a>
      
    
    </li></ul>
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Fragment-Based_Molecular_Replacement.html" class="md-nav__link">8. Fragment-Based Molecular Replacement</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Experimental_Phasing.html" class="md-nav__link">9. Experimental Phasing</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Density_Modification.html" class="md-nav__link">10. Density Modification</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Refinement_and_Model_Building.html" class="md-nav__link">11. Refinement and Model Building</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="COOT.html" class="md-nav__link">12. COOT</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Ligands.html" class="md-nav__link">13. Ligands</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Validation_Analysis_and_Deposition.html" class="md-nav__link">14. Validation, Analysis and Deposition</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Toolbox.html" class="md-nav__link">15. Toolbox</a>
      
    
    </li>
  </ul>
  

</nav>
              </div>
            </div>
          </div>
          <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
            <div class="md-sidebar__scrollwrap">
              <div class="md-sidebar__inner">
                
<nav class="md-nav md-nav--secondary">
    <label class="md-nav__title" for="__toc">Contents</label>
  <ul class="md-nav__list" data-md-scrollfix="">
        <li class="md-nav__item"><a href="#doc-task-ensembleprepmg--page-root" class="md-nav__link">7.1.8. Prepare MR Ensemble with CCP4mg</a><nav class="md-nav">
              <ul class="md-nav__list">
        <li class="md-nav__item"><a href="#parameters" class="md-nav__link">7.1.8.1. Parameters</a>
        </li>
        <li class="md-nav__item"><a href="#workflow" class="md-nav__link">7.1.8.2. Workflow</a>
        </li></ul>
            </nav>
        </li>
    
<li class="md-nav__item"><a class="md-nav__extra_link" href="_sources/doc.task.EnsemblePrepMG.rst.txt">Show Source</a> </li>

<li id="searchbox" class="md-nav__item"></li>

  </ul>
</nav>
              </div>
            </div>
          </div>
        
        <div class="md-content">
          <article class="md-content__inner md-typeset" role="main">
            
  
<h1 id="doc-task-ensembleprepmg--page-root"><span class="section-number">7.1.8. </span>Prepare MR Ensemble with CCP4mg<a class="headerlink" href="#doc-task-ensembleprepmg--page-root" title="Permalink to this headline">¶</a></h1>
<p>This task starts <a class="reference external" href="https://www.ccp4.ac.uk/MG/" rel="noopener noreferrer" target="_blank">CCP4mg</a> graphical viewer on your computer in a special mode, which allows you to find structural homologs to your target protein, trim and ensemble them interactively with full visual control in real-time. This task uses <a class="reference internal" href="doc.task.MrBump.html"><span class="doc">MrBump</span></a> software to make the <a class="reference internal" href="doc.task.Ensembler.html"><span class="doc">MR Ensemble</span></a> from given structural homologs.</p>
<p>MR Ensembles are obtained with an alignment and truncation procedure using homologues structures found in the PDB and <a class="reference external" href="https://doi.org/10.1038/s41586-021-03819-2" rel="noopener noreferrer" target="_blank">AFDB (AlphaFold DataBase)</a> with a sequence-based search.</p>
<p>Automatically generated ensembles may be further improved by manual selection of found homologs and removing unsuitable structural features. That can be especially useful in cases where homologous structures, found by <a class="reference external" href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002195" rel="noopener noreferrer" target="_blank">phmmer</a>, have multiple domains and no structural alignment can adequately superpose all domains simultaneously. For such proteins, the user may wish to specify which domain to use as the search model or to use more than one domain with different structural alignments. In other instances, only distant homologues may be found and it may be necessary to use only part of the structures in MR.</p>
<p>It may be convenient to group, or cluster, the homologs found on the basis of their structural similarity. This can be done in CCP4mg interface with the rainbow colouring of structures displayed, using their <a class="reference external" href="https://academic.oup.com/bioinformatics/article/21/7/951/268976" rel="noopener noreferrer" target="_blank">phmmer score</a>. The colouring scheme corresponds to that of the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1160169/" rel="noopener noreferrer" target="_blank">HHpred</a> web-application, with red indicating a high-scoring match through to blue at the lower end of the scale.</p>

<h2 id="parameters"><span class="section-number">7.1.8.1. </span>Parameters<a class="headerlink" href="#parameters" title="Permalink to this headline">¶</a></h2>
<p><strong>PDB sequence redundancy level</strong> specifies the highest acceptable degree of similarity between structural models in the ensemble. The task offers a choice of 100%, 95%, 90%, 70% and 50% redundancy levels.</p>
<p>At 100% redundancy level, all models with identical sequences, if found, are included in the ensemble. This option can be particularly useful for a protein that can adopt a range of different conformations or in cases where minor variations in the alignment between a set of very similar structures can guide the identification of the most conserved core residues.</p>
<p>At redundancy level lower than 100%, only one out of two or more models with sequence similarity higher than the redundancy level, can be used. In such cases, a model from crystal with higher resolution and lower R-factor is selected as a representative of the redundant subset of models.</p>
<p>The 95% redundancy level provides the most broad sampling of search models, while not allowing for models with identical and nearly identical sequences in the ensemble (thus filtering out models obtained from, <em>e.g.,</em> PDB depositions corresponding to the same protein with a series of ligands bound).</p>
<p><strong>Include structures from AFDB:</strong> this option allows to look for structural homologs, suitable for making MR search models, in <a class="reference external" href="https://doi.org/10.1038/s41586-021-03819-2" rel="noopener noreferrer" target="_blank">AFDB (AlphaFold DataBase)</a> (in addition to the PDB).</p>
<blockquote>
<div><p><strong>EBI-AFDB pLDDT cut-off:</strong> AlphaFold produces a per-residue estimate of its confidence score, called <strong>pLDDT</strong> on a scale from 0 - 100. This option sets the minimum level of <strong>pLDDT</strong> of selected residues.</p>
<ul class="simple">
<li><strong>pLDDT &gt; 90</strong>: residues are expected to be positioned with a high accuracy. These models should be suitable for any application that benefits from high accuracy (e.g. characterising binding sites).</li>
<li><strong>70 &lt; pLDDT &lt; 90</strong> residues are expected to be positioned reasonably well (generally, a good backbone prediction).</li>
<li><strong>50 &lt; pLDDT &lt; 70</strong> corresponds to residues modelled with low confidence, which should be treated with caution.</li>
<li><strong>pLDDT &lt; 50</strong> indicates residues that should not be interpreted. Such scores are a reasonably strong predictor of disorder, i.e. they suggest that the region is either unstructured in physiological conditions, or gets structured only as part of a complex.</li>
</ul>
</div></blockquote>
<p>You can search only the PDB or EBI-AFDB by unticking the relevant box in task interface.</p>


<h2 id="workflow"><span class="section-number">7.1.8.2. </span>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline">¶</a></h2>
<p>The task requires only amino acid sequence to prepare MR Ensemble with MrBump and CCP4mg.</p>
<blockquote>
<div><ul class="simple">
<li>homologous structures are found in PDB and/or AFDB by <a class="reference internal" href="doc.task.MrBump.html"><span class="doc">MrBump</span></a>, which uses <a class="reference external" href="https://www.ebi.ac.uk/Tools/hmmer/search/phmmer" rel="noopener noreferrer" target="_blank">phmmer</a> for scenning the corresponding sequence databases.</li>
<li>Based on the sequence alignment, search models are made from found homologs using <a class="reference external" href="https://doi.org/10.1107/S0907444910051218" rel="noopener noreferrer" target="_blank">Sculptor</a>, which can modify the the structures in three different ways: main-chain deletion, side-chain pruning and B-factor modification.</li>
<li>Prepared MR search models are superposed using <a class="reference internal" href="doc.task.Gesamt.html"><span class="doc">GESAMT</span></a>, and displayed in CCP4mg.</li>
<li>Finally, MR ensemble(s) need to be composed from prepared and superposed models. This is achieved by manual selection of models and pruning the resulting ensembles on the basis of r.m.s.d. between superposed residues. The models are selected using graphical controls in the <strong>show ranges view</strong> window, and outlier residues are removed using the <strong>GESAMT variance slider</strong>.</li>
<li>once suitable ensembles are created, they should be saved back to the Cloud, using <strong>File =&gt; Save all visible to the CCP4 Cloud</strong> menu item of CCP4mg.</li>
</ul>
</div></blockquote>
<p>Created ensembles can be used in both <a class="reference internal" href="doc.task.Phaser.html"><span class="doc">Phaser</span></a> and <a class="reference internal" href="doc.task.Molrep.html"><span class="doc">Molrep</span></a> tasks.</p>
<p><strong>Useful links</strong></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/MG/" rel="noopener noreferrer" target="_blank">CCP4mg webpage</a></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/MG/ccp4mg_help/tutorial/index.html" rel="noopener noreferrer" target="_blank">CCP4mg tutorials</a></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/MG/ccp4mg_help/index.html" rel="noopener noreferrer" target="_blank">CCP4mg documentation</a></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/html/mrbump.html" rel="noopener noreferrer" target="_blank">MrBUMP documentation</a></p>
<p><strong>REFERENCES</strong></p>
<p><a class="reference external" href="https://doi.org/10.1093/nar/gki408" rel="noopener noreferrer" target="_blank">Söding, J., Biegert, A., &amp; Lupas, A. N. (2005). The HHpred interactive server for protein homology detection and structure prediction. Nucleic acids research, 33(Web Server issue), W244–W248.</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S0907444911007281" rel="noopener noreferrer" target="_blank">McNicholas, S., Potterton, E., Wilson, K.S., Noble, M.E.M. (2011) Presenting your structures: the CCP4mg molecular-graphics software. Acta Cryst. D67: 386-394</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S2059798318003455" rel="noopener noreferrer" target="_blank">Keegan, R.M., McNicholas, S.J., Thomas, J.M.H., Simpkin, A.J., Simkovic, F., Uski, V., Ballard, C.C., Winn, M.D., Wilson, K.S., Rigden, D.J. (2018) Recent developments in MrBUMP: better search-model preparation, graphical interaction with search models, and solution improvement and assessment. Acta Cryst. D74: 167-182</a></p>
<p><a class="reference external" href="https://doi.org/10.1038/s41586-021-03819-2" rel="noopener noreferrer" target="_blank">Jumper, J., Evans, R., Pritzel, A. et al. Highly accurate protein structure prediction with AlphaFold. Nature 596, 583–589 (2021).</a></p>




          </article>
        </div>
      </div>
    </main>
  </div>
  <footer class="md-footer">
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
          
            <a href="doc.task.EnsemblePrepXYZ.html" title="7.1.7. Prepare MR Ensemble from Coordinate Data"
               class="md-flex md-footer-nav__link md-footer-nav__link--prev"
               rel="prev">
              <div class="md-flex__cell md-flex__cell--shrink">
                <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
              </div>
              <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
                <span class="md-flex__ellipsis">
                  <span
                      class="md-footer-nav__direction"> Previous </span> <span class="section-number">7.1.7. </span>Prepare MR Ensemble from Coordinate Data </span>
              </div>
            </a>
          
          
            <a href="doc.task.Phaser.html" title="7.2. Phaser"
               class="md-flex md-footer-nav__link md-footer-nav__link--next"
               rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span
                class="md-flex__ellipsis"> <span
                class="md-footer-nav__direction"> Next </span> <span class="section-number">7.2. </span>Phaser </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink"><i
                class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          
        </a>
        
      </nav>
    </div>
    <div class="md-footer-meta md-typeset">
      <div class="md-footer-meta__inner md-grid">
        <div class="md-footer-copyright">
            Created using
            <a href="http://www.sphinx-doc.org/">Sphinx</a> 4.0.1.
             and
            <a href="https://github.com/bashtage/sphinx-material/">Material for
              Sphinx</a>
        </div>
      </div>
    </div>
  </footer>
  <script src="_static/javascripts/application.js"></script>
  <script>app.initialize({version: "1.0.4", url: {base: ".."}})</script>
  </body>
</html>