
<!DOCTYPE html>

<html>
  <head>

    <script async src="https://www.googletagmanager.com/gtag/js?id=G-FCVD2T7BGM"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'G-FCVD2T7BGM');
    </script>
    
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>6.1.3. Prepare MR Model(s) from Alignment data &#8212; CCP4 Cloud 1.7 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/haiku.css" />
    <link rel="stylesheet" type="text/css" href="_static/contentui.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/contentui.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="6.1.4. Prepare Complex Model" href="doc.task.ModelPrepMC.html" />
    <link rel="prev" title="6.1.2. Prepare MR Model(s) from Coordinate data" href="doc.task.ModelPrepXYZ.html" /> 
  </head><body>
      <div class="header" role="banner"><h1 class="heading"><a href="index.html">
          <span>CCP4 Cloud 1.7 documentation</span></a></h1>
        <h2 class="heading"><span>6.1.3. Prepare MR Model(s) from Alignment data</span></h2>
      </div>
      <div class="topnav" role="navigation" aria-label="top navigation">
      
        <p>
        «&#160;&#160;<a href="doc.task.ModelPrepXYZ.html"><span class="section-number">6.1.2. </span>Prepare MR Model(s) from Coordinate data</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="doc.task.ModelPrepMC.html"><span class="section-number">6.1.4. </span>Prepare Complex Model</a>&#160;&#160;»
        </p>

      </div>
      <div class="content" role="main">
        
        
  <div class="section" id="prepare-mr-model-s-from-alignment-data">
<h1><span class="section-number">6.1.3. </span>Prepare MR Model(s) from Alignment data<a class="headerlink" href="#prepare-mr-model-s-from-alignment-data" title="Permalink to this headline">¶</a></h1>
<p>Usually, sufficiently high amino acid sequence similarity correlates well with structural similarity, which fact is used for finding suitable structure homologs from PDB or other data bases. This task uses the <a class="reference external" href="https://toolkit.tuebingen.mpg.de/tools/hhpred" rel="noopener noreferrer" target="_blank">HHpred alignment</a> for identifying PDB entries, suitable for making MR search models. HHpred is and advanced bioinformatics tool that can detect structural homology even when sequence similarity is at the bottom of the twilight zone and even below that. The sequence alignment generated in the search step can be used as a guide for the pruning of the homologue.</p>
<p>The task requires a preliminary import of a <code class="docutils literal notranslate"><span class="pre">.hhr</span></code> file with the results of HHpred search done at the <a class="reference external" href="https://toolkit.tuebingen.mpg.de/tools/hhpred" rel="noopener noreferrer" target="_blank">HHpred</a> web-site using the target sequence.</p>
<p>The atomic coordinates of PDB entries listed in HHpred results are automatically trimmed according to the selection ranges given in HHpred output. The results are usually modified (pruned) in order to remove parts that do not correspond to the target sequence. A few modification protocols are available:</p>
<blockquote>
<div><ul class="simple">
<li><strong>Unmodified</strong>: atomic coordinates are not altered in any way</li>
<li><strong>Clip</strong>: water molecules and ligands are removed</li>
<li><strong>Molrep</strong>: modification protocol, based on the comparison of the target sequence and atomic coordinates, used by
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2394799/" rel="noopener noreferrer" target="_blank">Molrep</a></li>
<li><strong>Sculptor</strong>: modification protocols, based on the comparison of the target sequence and atomic coordinates, provided by
<code class="docutils literal notranslate"><span class="pre">Phaser.Sculptor</span></code> (<em>cf.</em> <a class="reference external" href="https://www.phaser.cimr.cam.ac.uk/index.php/Molecular_Replacement" rel="noopener noreferrer" target="_blank">here</a>)</li>
<li><strong>Chainsaw</strong>: modification protocols, based on the comparison of the target sequence and atomic coordinates, provided by
<a class="reference external" href="https://doi.org/10.1107/S0021889808006985" rel="noopener noreferrer" target="_blank">Chainsaw</a></li>
<li><strong>Polyalanine</strong>: all residues are replaced with polyalanines</li>
</ul>
</div></blockquote>
<p>The resulting models can be used in both <a class="reference internal" href="doc.task.Phaser.html"><span class="doc">Phaser</span></a> and <a class="reference internal" href="doc.task.Molrep.html"><span class="doc">Molrep</span></a> tasks, but the outcome may be different, especially in difficult cases.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">the task usually produces several MR models with a range of similarity to target sequence. Although higher sequence similarity implies higher success or Molecular Replacement, this is not guaranteed.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">combining several models, corresponding to the same target sequence, into MR Ensemble(s) may increase chances for success. MR ensembles may be prepared from individual models using the <a class="reference internal" href="doc.task.Ensembler.html"><span class="doc">MR Ensembles</span></a> task</p>
</div>
<p><strong>REFERENCES</strong></p>
<p><a class="reference external" href="http://www.sciencedirect.com/science/article/pii/S0022283617305879" rel="noopener noreferrer" target="_blank">A Completely Reimplemented MPI Bioinformatics Toolkit with a New HHpred Server at its Core.
Zimmermann L, Stephens A, Nam SZ, Rau D, Kübler J, Lozajic M, Gabler F, Söding J, Lupas AN, Alva V. J Mol Biol. 2018 Jul 20. S0022-2836(17)30587-9.</a></p>
<p><a class="reference external" href="https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpbi.108" rel="noopener noreferrer" target="_blank">Protein Sequence Analysis Using the MPI Bioinformatics Toolkit.
Gabler F, Nam SZ, Till S, Mirdita M, Steinegger M, Söding J, Lupas AN, Alva V. Curr Protoc Bioinformatics. 2020 Dec;72(1):e108. doi: 10.1002/cpbi.108.</a></p>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/15531603" rel="noopener noreferrer" target="_blank">Söding J. (2005) Protein homology detection by HMM-HMM comparison. Bioinformatics 21: 951-960. PMID: 15531603</a></p>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19626712" rel="noopener noreferrer" target="_blank">Hildebrand A., Remmert A., Biegert A., Söding J. (2009). Fast and accurate automatic structure prediction with HHpred. Proteins 77(Suppl 9):128-32. doi: 10.1002/prot.22499. PMID: 19626712</a></p>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/26496371" rel="noopener noreferrer" target="_blank">Meier A., Söding J. (2015). Automatic Prediction of Protein 3D Structures by Probabilistic Multi-template Homology Modeling. PLoS Comput Biol. 11(10):e1004343. doi: 10.1371/journal.pcbi.1004343. PMID: 26496371</a></p>
</div>


      </div>
      <div class="bottomnav" role="navigation" aria-label="bottom navigation">
      
        <p>
        «&#160;&#160;<a href="doc.task.ModelPrepXYZ.html"><span class="section-number">6.1.2. </span>Prepare MR Model(s) from Coordinate data</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="doc.task.ModelPrepMC.html"><span class="section-number">6.1.4. </span>Prepare Complex Model</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer" role="contentinfo">
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.0.1.
    </div>
  </body>
</html>