

<!DOCTYPE html>
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>6.1.8. Prepare MR Ensemble with CCP4mg &mdash; CCP4 Cloud 1.7 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.min.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/custom.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.min.css" type="text/css" />
  <link rel="stylesheet" href="_static/contentui.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="6.2. Phaser" href="doc.task.Phaser.html" />
    <link rel="prev" title="6.1.7. Prepare MR Ensemble from Coordinate Data" href="doc.task.EnsemblePrepXYZ.html" /> 

</head>

<body>
    <header>
        <div class="container">
            <a class="site-nav-toggle hidden-lg-up"><i class="icon-menu"></i></a>
            <a class="site-title" href="index.html">
                CCP4 Cloud
            </a>
        </div>
    </header>


<div class="breadcrumbs-outer hidden-xs-down">
    <div class="container">
        















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="breadcrumbs">
    
      <li><a href="index.html">Docs</a></li>
        
          <li><a href="Molecular_Replacement.html"><span class="section-number">6. </span>Molecular Replacement</a></li>
        
          <li><a href="doc.task.MR_model_preparation.html"><span class="section-number">6.1. </span>Molecular Replacement Models and Ensembles</a></li>
        
      <li><span class="section-number">6.1.8. </span>Prepare MR Ensemble with CCP4mg</li>
    
    
      <li class="breadcrumbs-aside">
        
            
            <a href="_sources/doc.task.EnsemblePrepMG.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>
</div>
    </div>
</div>
    <div class="main-outer">
        <div class="container">
            <div class="row">
                <div class="col-12 col-lg-3 site-nav">
                    
<div role="search">
    <form class="search" action="search.html" method="get">
        <div class="icon-input">
            <input type="text" name="q" placeholder="Search" />
            <span class="icon-search"></span>
        </div>
        <input type="submit" value="Go" class="d-hidden" />
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
    </form>
</div>
                    <div class="site-nav-tree">
                        
                            
                            
                                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Workflows.html">1. Automatic workflows</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.WFlowAEP.html">1.1. Automatic Experimental Phasing Workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.WFlowAMR.html">1.2. Automatic Molecular Replacement Workflow</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Data_Import.html">2. Data Import</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Import.html">2.1. Data Import</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ImportSeqCP.html">2.2. Import Sequence(s) by Copy-Paste</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ImportPDB.html">2.3. Import from PDB</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CloudImport.html">2.4. Import from Cloud Storage</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ImportReplace.html">2.5. Import &amp; Replace</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Migrate.html">2.6. Hop on CCP4 Cloud</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Data_Processing.html">3. Data Processing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Xia2.html">3.1. Automatic Image Processing with Xia-2</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.DUI.html">3.2. Interactive Image Processing with DIALS/DUI</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.iMosflm.html">3.3. Interactive Image Processing with iMosflm</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Aimless.html">3.4. Data Reduction with Aimless</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Asymmetric_Unit_and_Structure_Revision.html">4. Asymmetric Unit and Structure Revision</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Asymmetric_Unit_and_Structure_Revision.html#asymmetric-unit">4.1. Asymmetric Unit</a></li>
<li class="toctree-l2"><a class="reference internal" href="Asymmetric_Unit_and_Structure_Revision.html#structure-revision">4.2. Structure Revision</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Automated_Molecular_Replacement.html">5. Automated Molecular Replacement</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Morda.html">5.1. Morda</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.MrBump.html">5.2. MrBump</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Balbes.html">5.3. Balbes</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Simbad.html">5.4. Simbad</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Ample.html">5.5. Ample</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="Molecular_Replacement.html">6. Molecular Replacement</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="doc.task.MR_model_preparation.html">6.1. Molecular Replacement Models and Ensembles</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Phaser.html">6.2. Phaser</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Molrep.html">6.3. Molrep</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Fragment-Based_Molecular_Replacement.html">7. Fragment-Based Molecular Replacement</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Arcimboldo.html">7.1. Fragment Molecular Replacement with Arcimboldo</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ArcimboldoBorges.html">7.2. Arcimboldo Borges</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ArcimboldoShredder.html">7.3. Arcimboldo Shredder</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Experimental_Phasing.html">8. Experimental Phasing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Crank2.html">8.1. Crank-2</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html">8.2. SHELXCD</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html#id1">8.3. SHELXE</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Phaser.html">6.2. Phaser</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Density_Modification.html">9. Density Modification</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Parrot.html">9.1. Density Modification with Parrot</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Acorn.html">9.2. Phase Refinement and Dynamic Density Modification with ACORN</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html">8.2. SHELXCD</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html#id1">8.3. SHELXE</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Refinement_and_Model_Building.html">10. Refinement and Model Building</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Refmac.html">10.1. REFMAC5</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Buster.html">10.2. Buster Refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Lorestr.html">10.3. Low-Resolution Refinement with Lorestr</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CCP4Build.html">10.4. Automatic Model Building with CCP4Build</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Buccaneer.html">10.5. Automatic Model Building with Buccaneer</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ArpWarp.html">10.6. Automated Model Building with Arp/wArp</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ModelCraft.html">10.7. Automatic Model Building with Modelcraft</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Nautilus.html">10.8. Automatic Model Building of RNA/DNA with Nautilus</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Dimple.html">10.9. Dimple</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CombStructure.html">10.10. Comb Structure with Coot/Refmac</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="COOT.html">11. COOT</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CootMB.html">11.1. Model Building with Coot</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CootCE.html">11.2. Edit Coordinates with Coot</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Ligands.html">12. Ligands</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.MakeLigand.html">12.1. Make Ligand with Acedrg</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.FitLigand.html">12.2. Find and Fit Ligand with Coot</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.FitWaters.html">12.3. Find and Fit Waters with Coot</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Validation_Analysis_and_Deposition.html">13. Validation, Analysis and Deposition</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Zanuda.html">13.1. Space Group Validation with Zanuda</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Privateer.html">13.2. Validation of carbohydrate structures with Privateer</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.PISA.html">13.3. Surface, Interfaces and Assembly Analysis with PISA</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Deposition.html">13.4. Prepare data for deposition</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Toolbox.html">14. Toolbox</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Auspex.html">14.1. Reflection data diagnostics with Auspex plots</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SRF.html">14.2. Self-Rotation Function Analysis with Molrep</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CrosSec.html">14.3. X-ray cross sections and anomalous scattering factors</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.XyzUtils.html">14.4. Coordinate Utilities</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Gemmi.html">14.5. Gemmi</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Gesamt.html">14.6. Pairwise Structural Alignment with GESAMT</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.LsqKab.html">14.7. Structure Superposition with LsqKab</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SeqAlign.html">14.8. Sequence Alignment with ClustalW</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SymMatch.html">14.9. Symmetry Match to Reference Structure with CSymMatch</a></li>
</ul>
</li>
</ul>

                            
                        
                    </div>
                </div>
                <div class="col-12 col-lg-9">
                    <div class="document">
                        
                            
  <div class="section" id="prepare-mr-ensemble-with-ccp4mg">
<h1><span class="section-number">6.1.8. </span>Prepare MR Ensemble with CCP4mg<a class="headerlink" href="#prepare-mr-ensemble-with-ccp4mg" title="Permalink to this headline">¶</a></h1>
<p>This task starts <a class="reference external" href="https://www.ccp4.ac.uk/MG/" rel="noopener noreferrer" target="_blank">CCP4mg</a> graphical viewer on your computer in a special mode, which allows you to find structural homologs to your target protein, trim and ensemble them interactively with full visual control in real-time. This task uses <a class="reference internal" href="doc.task.MrBump.html"><span class="doc">MrBump</span></a> software to make the <a class="reference internal" href="doc.task.Ensembler.html"><span class="doc">MR Ensemble</span></a> from given structural homologs.</p>
<p>MR Ensembles are obtained with an alignment and truncation procedure using homologues structures found in the PDB and <a class="reference external" href="https://doi.org/10.1038/s41586-021-03819-2" rel="noopener noreferrer" target="_blank">AFDB (AlphaFold DataBase)</a> with a sequence-based search.</p>
<p>Automatically generated ensembles may be further improved by manual selection of found homologs and removing unsuitable structural features. That can be especially useful in cases where homologous structures, found by <a class="reference external" href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002195" rel="noopener noreferrer" target="_blank">phmmer</a>, have multiple domains and no structural alignment can adequately superpose all domains simultaneously. For such proteins, the user may wish to specify which domain to use as the search model or to use more than one domain with different structural alignments. In other instances, only distant homologues may be found and it may be necessary to use only part of the structures in MR.</p>
<p>It may be convenient to group, or cluster, the homologs found on the basis of their structural similarity. This can be done in CCP4mg interface with the rainbow colouring of structures displayed, using their <a class="reference external" href="https://academic.oup.com/bioinformatics/article/21/7/951/268976" rel="noopener noreferrer" target="_blank">phmmer score</a>. The colouring scheme corresponds to that of the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1160169/" rel="noopener noreferrer" target="_blank">HHpred</a> web-application, with red indicating a high-scoring match through to blue at the lower end of the scale.</p>
<div class="section" id="parameters">
<h2><span class="section-number">6.1.8.1. </span>Parameters<a class="headerlink" href="#parameters" title="Permalink to this headline">¶</a></h2>
<p><strong>PDB sequence redundancy level</strong> specifies the highest acceptable degree of similarity between structural models in the ensemble. The task offers a choice of 100%, 95%, 90%, 70% and 50% redundancy levels.</p>
<p>At 100% redundancy level, all models with identical sequences, if found, are included in the ensemble. This option can be particularly useful for a protein that can adopt a range of different conformations or in cases where minor variations in the alignment between a set of very similar structures can guide the identification of the most conserved core residues.</p>
<p>At redundancy level lower than 100%, only one out of two or more models with sequence similarity higher than the redundancy level, can be used. In such cases, a model from crystal with higher resolution and lower R-factor is selected as a representative of the redundant subset of models.</p>
<p>The 95% redundancy level provides the most broad sampling of search models, while not allowing for models with identical and nearly identical sequences in the ensemble (thus filtering out models obtained from, <em>e.g.,</em> PDB depositions corresponding to the same protein with a series of ligands bound).</p>
<p><strong>Include structures from AFDB:</strong> this option allows to look for structural homologs, suitable for making MR search models, in <a class="reference external" href="https://doi.org/10.1038/s41586-021-03819-2" rel="noopener noreferrer" target="_blank">AFDB (AlphaFold DataBase)</a> (in addition to the PDB).</p>
<blockquote>
<div><p><strong>EBI-AFDB pLDDT cut-off:</strong> AlphaFold produces a per-residue estimate of its confidence score, called <strong>pLDDT</strong> on a scale from 0 - 100. This option sets the minimum level of <strong>pLDDT</strong> of selected residues.</p>
<ul class="simple">
<li><strong>pLDDT &gt; 90</strong>: residues are expected to be positioned with a high accuracy. These models should be suitable for any application that benefits from high accuracy (e.g. characterising binding sites).</li>
<li><strong>70 &lt; pLDDT &lt; 90</strong> residues are expected to be positioned reasonably well (generally, a good backbone prediction).</li>
<li><strong>50 &lt; pLDDT &lt; 70</strong> corresponds to residues modelled with low confidence, which should be treated with caution.</li>
<li><strong>pLDDT &lt; 50</strong> indicates residues that should not be interpreted. Such scores are a reasonably strong predictor of disorder, i.e. they suggest that the region is either unstructured in physiological conditions, or gets structured only as part of a complex.</li>
</ul>
</div></blockquote>
<p>You can search only the PDB or EBI-AFDB by unticking the relevant box in task interface.</p>
</div>
<div class="section" id="workflow">
<h2><span class="section-number">6.1.8.2. </span>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline">¶</a></h2>
<p>The task requires only amino acid sequence to prepare MR Ensemble with MrBump and CCP4mg.</p>
<blockquote>
<div><ul class="simple">
<li>homologous structures are found in PDB and/or AFDB by <a class="reference internal" href="doc.task.MrBump.html"><span class="doc">MrBump</span></a>, which uses <a class="reference external" href="https://www.ebi.ac.uk/Tools/hmmer/search/phmmer" rel="noopener noreferrer" target="_blank">phmmer</a> for scenning the corresponding sequence databases.</li>
<li>Based on the sequence alignment, search models are made from found homologs using <a class="reference external" href="https://doi.org/10.1107/S0907444910051218" rel="noopener noreferrer" target="_blank">Sculptor</a>, which can modify the the structures in three different ways: main-chain deletion, side-chain pruning and B-factor modification.</li>
<li>Prepared MR search models are superposed using <a class="reference internal" href="doc.task.Gesamt.html"><span class="doc">GESAMT</span></a>, and displayed in CCP4mg.</li>
<li>Finally, MR ensemble(s) need to be composed from prepared and superposed models. This is achieved by manual selection of models and pruning the resulting ensembles on the basis of r.m.s.d. between superposed residues. The models are selected using graphical controls in the <strong>show ranges view</strong> window, and outlier residues are removed using the <strong>GESAMT variance slider</strong>.</li>
<li>once suitable ensembles are created, they should be saved back to the Cloud, using <strong>File =&gt; Save all visible to the CCP4 Cloud</strong> menu item of CCP4mg.</li>
</ul>
</div></blockquote>
<p>Created ensembles can be used in both <a class="reference internal" href="doc.task.Phaser.html"><span class="doc">Phaser</span></a> and <a class="reference internal" href="doc.task.Molrep.html"><span class="doc">Molrep</span></a> tasks.</p>
<p><strong>Useful links</strong></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/MG/" rel="noopener noreferrer" target="_blank">CCP4mg webpage</a></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/MG/ccp4mg_help/tutorial/index.html" rel="noopener noreferrer" target="_blank">CCP4mg tutorials</a></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/MG/ccp4mg_help/index.html" rel="noopener noreferrer" target="_blank">CCP4mg documentation</a></p>
<p><a class="reference external" href="https://www.ccp4.ac.uk/html/mrbump.html" rel="noopener noreferrer" target="_blank">MrBUMP documentation</a></p>
<p><strong>REFERENCES</strong></p>
<p><a class="reference external" href="https://doi.org/10.1093/nar/gki408" rel="noopener noreferrer" target="_blank">Söding, J., Biegert, A., &amp; Lupas, A. N. (2005). The HHpred interactive server for protein homology detection and structure prediction. Nucleic acids research, 33(Web Server issue), W244–W248.</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S0907444911007281" rel="noopener noreferrer" target="_blank">McNicholas, S., Potterton, E., Wilson, K.S., Noble, M.E.M. (2011) Presenting your structures: the CCP4mg molecular-graphics software. Acta Cryst. D67: 386-394</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S2059798318003455" rel="noopener noreferrer" target="_blank">Keegan, R.M., McNicholas, S.J., Thomas, J.M.H., Simpkin, A.J., Simkovic, F., Uski, V., Ballard, C.C., Winn, M.D., Wilson, K.S., Rigden, D.J. (2018) Recent developments in MrBUMP: better search-model preparation, graphical interaction with search models, and solution improvement and assessment. Acta Cryst. D74: 167-182</a></p>
<p><a class="reference external" href="https://doi.org/10.1038/s41586-021-03819-2" rel="noopener noreferrer" target="_blank">Jumper, J., Evans, R., Pritzel, A. et al. Highly accurate protein structure prediction with AlphaFold. Nature 596, 583–589 (2021).</a></p>
</div>
</div>


                        
                    </div>
                </div>
            </div>
        </div>
    </div>    


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.7',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/contentui.js"></script>
    <script type="text/javascript" src="_static/js/theme.js"></script>
  
    <div class="footer" role="contentinfo">
        <div class="container">
        Created using <a href="http://sphinx-doc.org/">Sphinx</a> 4.0.1.
        </div>
    </div>  

</body>
</html>