
<!DOCTYPE html>

<html>
  <head>

    <script async src="https://www.googletagmanager.com/gtag/js?id=G-FCVD2T7BGM"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'G-FCVD2T7BGM');
    </script>
    
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>6.1.1. Find and prepare MR models with MrParse &#8212; CCP4 Cloud 1.7 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/haiku.css" />
    <link rel="stylesheet" type="text/css" href="_static/contentui.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/contentui.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="6.1.2. Prepare MR Model(s) from Coordinate data" href="doc.task.ModelPrepXYZ.html" />
    <link rel="prev" title="6.1. Molecular Replacement Models and Ensembles" href="doc.task.MR_model_preparation.html" /> 
  </head><body>
      <div class="header" role="banner"><h1 class="heading"><a href="index.html">
          <span>CCP4 Cloud 1.7 documentation</span></a></h1>
        <h2 class="heading"><span>6.1.1. Find and prepare MR models with MrParse</span></h2>
      </div>
      <div class="topnav" role="navigation" aria-label="top navigation">
      
        <p>
        «&#160;&#160;<a href="doc.task.MR_model_preparation.html"><span class="section-number">6.1. </span>Molecular Replacement Models and Ensembles</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="doc.task.ModelPrepXYZ.html"><span class="section-number">6.1.2. </span>Prepare MR Model(s) from Coordinate data</a>&#160;&#160;»
        </p>

      </div>
      <div class="content" role="main">
        
        
  <div class="section" id="find-and-prepare-mr-models-with-mrparse">
<h1><span class="section-number">6.1.1. </span>Find and prepare MR models with MrParse<a class="headerlink" href="#find-and-prepare-mr-models-with-mrparse" title="Permalink to this headline">¶</a></h1>
<p>MrParse identifies structural homologs, suitable for making MR search models, in PDB and AFDB (AlphaFold Data Base). Relevant structures are selected on the basis of sequence alignment with <a class="reference external" href="http://hmmer.org/" rel="noopener noreferrer" target="_blank">PHMMER</a> software. PHMMER implements methods using probabilistic models called profile hidden Markov Models, and is considered to be one of the most sensitive sequence-comparison software.</p>
<p>Found structure homologs are then trimmed to maximising their resemblance to target sequence and increasing chances for success in Molecular Replacement. If reflection data set is provided (recommended), MrParse uses the Phaser software to calculate eLLG scores for MR models found <a class="reference external" href="https://doi.org/10.1107/S2059798318004357" rel="noopener noreferrer" target="_blank">[Oeffner, 2018]</a>. MR models with higher eLLG are almost certainly better for Molecular Replacement than models with lower eLLG scores and should be tried first.</p>
<p>MR search models, prepared by MrParse task, may be used in Molecular Replacement tasks (Phaser-MR or Molrep) directly, or assembled in  <a class="reference internal" href="doc.task.Ensembler.html"><span class="doc">MR Ensembles</span></a> before phasing, which may increase chances for success.</p>
<p><strong>REFERENCES</strong></p>
<p><a class="reference external" href="https://www.cambridge.org/core/books/biological-sequence-analysis/921BB7B78B745198829EF96BC7E0F29D" rel="noopener noreferrer" target="_blank">Durbin, R., Eddy, S. R., Krogh, A., and Mitchison, G. J. (1998) Biological Sequence Analysis: Probabilistic Models of Proteins and Nucleic Acids. Cambridge University Press ;</a></p>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/15531603" rel="noopener noreferrer" target="_blank">Söding J. (2005) Protein homology detection by HMM-HMM comparison. Bioinformatics 21: 951-960. PMID: 15531603</a></p>
<p><a class="reference external" href="http://www.sciencedirect.com/science/article/pii/S0022283617305879" rel="noopener noreferrer" target="_blank">Zimmermann L, Stephens A, Nam SZ, Rau D, Kübler J, Lozajic M, Gabler F, Söding J, Lupas AN. (2018) A Completely Reimplemented MPI Bioinformatics Toolkit with a New HHpred Server at its Core. Alva V. J Mol Biol. S0022-2836(17)30587-9.</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S2059798318004357" rel="noopener noreferrer" target="_blank">Oeffner, R. D., Afonine, P. V., Millán, C., Sammito, M., Usón, I., Read, R. J., &amp; McCoy, A. J. (2018). On the application of the expected log-likelihood gain to decision making in molecular replacement. Acta crystallographica. Section D, Structural biology, 74(Pt 4), 245–255.</a></p>
</div>


      </div>
      <div class="bottomnav" role="navigation" aria-label="bottom navigation">
      
        <p>
        «&#160;&#160;<a href="doc.task.MR_model_preparation.html"><span class="section-number">6.1. </span>Molecular Replacement Models and Ensembles</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="doc.task.ModelPrepXYZ.html"><span class="section-number">6.1.2. </span>Prepare MR Model(s) from Coordinate data</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer" role="contentinfo">
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.0.1.
    </div>
  </body>
</html>