

<!DOCTYPE html>
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>7.3. Arcimboldo Shredder &mdash; CCP4 Cloud 1.7 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.min.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/custom.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.min.css" type="text/css" />
  <link rel="stylesheet" href="_static/contentui.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="8. Experimental Phasing" href="Experimental_Phasing.html" />
    <link rel="prev" title="7.2. Arcimboldo Borges" href="doc.task.ArcimboldoBorges.html" /> 

</head>

<body>
    <header>
        <div class="container">
            <a class="site-nav-toggle hidden-lg-up"><i class="icon-menu"></i></a>
            <a class="site-title" href="index.html">
                CCP4 Cloud
            </a>
        </div>
    </header>


<div class="breadcrumbs-outer hidden-xs-down">
    <div class="container">
        















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="breadcrumbs">
    
      <li><a href="index.html">Docs</a></li>
        
          <li><a href="Fragment-Based_Molecular_Replacement.html"><span class="section-number">7. </span>Fragment-Based Molecular Replacement</a></li>
        
      <li><span class="section-number">7.3. </span>Arcimboldo Shredder</li>
    
    
      <li class="breadcrumbs-aside">
        
            
            <a href="_sources/doc.task.ArcimboldoShredder.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>
</div>
    </div>
</div>
    <div class="main-outer">
        <div class="container">
            <div class="row">
                <div class="col-12 col-lg-3 site-nav">
                    
<div role="search">
    <form class="search" action="search.html" method="get">
        <div class="icon-input">
            <input type="text" name="q" placeholder="Search" />
            <span class="icon-search"></span>
        </div>
        <input type="submit" value="Go" class="d-hidden" />
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
    </form>
</div>
                    <div class="site-nav-tree">
                        
                            
                            
                                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Workflows.html">1. Automatic workflows</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.WFlowAEP.html">1.1. Automatic Experimental Phasing Workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.WFlowAMR.html">1.2. Automatic Molecular Replacement Workflow</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Data_Import.html">2. Data Import</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Import.html">2.1. Data Import</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ImportSeqCP.html">2.2. Import Sequence(s) by Copy-Paste</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ImportPDB.html">2.3. Import from PDB</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CloudImport.html">2.4. Import from Cloud Storage</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ImportReplace.html">2.5. Import &amp; Replace</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Migrate.html">2.6. Hop on CCP4 Cloud</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Data_Processing.html">3. Data Processing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Xia2.html">3.1. Automatic Image Processing with Xia-2</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.DUI.html">3.2. Interactive Image Processing with DIALS/DUI</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.iMosflm.html">3.3. Interactive Image Processing with iMosflm</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Aimless.html">3.4. Data Reduction with Aimless</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Asymmetric_Unit_and_Structure_Revision.html">4. Asymmetric Unit and Structure Revision</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Asymmetric_Unit_and_Structure_Revision.html#asymmetric-unit">4.1. Asymmetric Unit</a></li>
<li class="toctree-l2"><a class="reference internal" href="Asymmetric_Unit_and_Structure_Revision.html#structure-revision">4.2. Structure Revision</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Automated_Molecular_Replacement.html">5. Automated Molecular Replacement</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Morda.html">5.1. Morda</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.MrBump.html">5.2. MrBump</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Balbes.html">5.3. Balbes</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Simbad.html">5.4. Simbad</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Ample.html">5.5. Ample</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Molecular_Replacement.html">6. Molecular Replacement</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.MR_model_preparation.html">6.1. Molecular Replacement Models and Ensembles</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Phaser.html">6.2. Phaser</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Molrep.html">6.3. Molrep</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="Fragment-Based_Molecular_Replacement.html">7. Fragment-Based Molecular Replacement</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="doc.task.Arcimboldo.html">7.1. Fragment Molecular Replacement with Arcimboldo</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ArcimboldoBorges.html">7.2. Arcimboldo Borges</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">7.3. Arcimboldo Shredder</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Experimental_Phasing.html">8. Experimental Phasing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Crank2.html">8.1. Crank-2</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html">8.2. SHELXCD</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html#id1">8.3. SHELXE</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Phaser.html">6.2. Phaser</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Density_Modification.html">9. Density Modification</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Parrot.html">9.1. Density Modification with Parrot</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Acorn.html">9.2. Phase Refinement and Dynamic Density Modification with ACORN</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html">8.2. SHELXCD</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SHELX.html#id1">8.3. SHELXE</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Refinement_and_Model_Building.html">10. Refinement and Model Building</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Refmac.html">10.1. REFMAC5</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Buster.html">10.2. Buster Refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Lorestr.html">10.3. Low-Resolution Refinement with Lorestr</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CCP4Build.html">10.4. Automatic Model Building with CCP4Build</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Buccaneer.html">10.5. Automatic Model Building with Buccaneer</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ArpWarp.html">10.6. Automated Model Building with Arp/wArp</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.ModelCraft.html">10.7. Automatic Model Building with Modelcraft</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Nautilus.html">10.8. Automatic Model Building of RNA/DNA with Nautilus</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Dimple.html">10.9. Dimple</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CombStructure.html">10.10. Comb Structure with Coot/Refmac</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="COOT.html">11. COOT</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CootMB.html">11.1. Model Building with Coot</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CootCE.html">11.2. Edit Coordinates with Coot</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Ligands.html">12. Ligands</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.MakeLigand.html">12.1. Make Ligand with Acedrg</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.FitLigand.html">12.2. Find and Fit Ligand with Coot</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.FitWaters.html">12.3. Find and Fit Waters with Coot</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Validation_Analysis_and_Deposition.html">13. Validation, Analysis and Deposition</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Zanuda.html">13.1. Space Group Validation with Zanuda</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Privateer.html">13.2. Validation of carbohydrate structures with Privateer</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.PISA.html">13.3. Surface, Interfaces and Assembly Analysis with PISA</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Deposition.html">13.4. Prepare data for deposition</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Toolbox.html">14. Toolbox</a><ul>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Auspex.html">14.1. Reflection data diagnostics with Auspex plots</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SRF.html">14.2. Self-Rotation Function Analysis with Molrep</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.CrosSec.html">14.3. X-ray cross sections and anomalous scattering factors</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.XyzUtils.html">14.4. Coordinate Utilities</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Gemmi.html">14.5. Gemmi</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.Gesamt.html">14.6. Pairwise Structural Alignment with GESAMT</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.LsqKab.html">14.7. Structure Superposition with LsqKab</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SeqAlign.html">14.8. Sequence Alignment with ClustalW</a></li>
<li class="toctree-l2"><a class="reference internal" href="doc.task.SymMatch.html">14.9. Symmetry Match to Reference Structure with CSymMatch</a></li>
</ul>
</li>
</ul>

                            
                        
                    </div>
                </div>
                <div class="col-12 col-lg-9">
                    <div class="document">
                        
                            
  <div class="section" id="arcimboldo-shredder">
<h1><span class="section-number">7.3. </span>Arcimboldo Shredder<a class="headerlink" href="#arcimboldo-shredder" title="Permalink to this headline">¶</a></h1>
<p><a class="reference external" href="http://chango.ibmb.csic.es/shredder" rel="noopener noreferrer" target="_blank">Arcimboldo Shredder</a> uses fragments derived from distant homologs based on the experimental data, instead of by sequence similarity. In this approach, the initial template is systematically shredded, and fragments are scored against each unique solution of the rotation function. Results are combined into a score per residue and the template is trimmed accordingly.</p>
<p>The Arcimboldo Shredder performs parallel and systematic molecular-replacement searches using a library of small structure motifs derived from a homologous structure and analyses the results to extract information from the persistence of solutions for different fragments among the noisy rotation-function results from Phaser.</p>
<p>Potential molecular-replacement solutions are passed to SHELXE for density modification and model building, with the prospect that any correctly placed fragments can be expanded into a full structure.</p>
<div class="section" id="input">
<h2><span class="section-number">7.3.1. </span>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h2>
<div class="section" id="shredder-mode">
<h3><span class="section-number">7.3.1.1. </span>Shredder mode<a class="headerlink" href="#shredder-mode" title="Permalink to this headline">¶</a></h3>
<p>In the <strong>sequential</strong> mode, the initial template is systematically shredded and fragments are scored against each unique solution of the rotation function. Results are combined into a score per residue and the template is trimmed accordingly. Models are selected by optimization of PHASER’s rotation function and trimming in SHELXE. Sequential shredding can improve models where the high average deviation from the target is owing to dissimilar or flexible regions.</p>
<p>The <strong>spherical</strong> mode generates model, starting from a distant homolog template that is annotated in terms of secondary and tertiary structure, a set of compact search models used as a library in ARCIMBOLDO BORGES, profiting from specific options in PHASER and SHELXE for model improvement. Spherical shredding can improve models where deviations of the template from the target are evenly distributed among a similar fold.</p>
</div>
<div class="section" id="perform-gyre-refinement">
<h3><span class="section-number">7.3.1.2. </span>Perform gyre refinement<a class="headerlink" href="#perform-gyre-refinement" title="Permalink to this headline">¶</a></h3>
<p>The maximum-likelihood GYRE method implemented in Phaser for optimizing the orientation and relative position of rigid-body fragments of a model after the orientation of the model has been identified, but before the model has been positioned in the unit cell. It performs the refinement of rigid-body groups against the rotation function target.</p>
<p>Atoms with different chain identifiers within an ensemble will be treated as independent rigid groups, refining their rotation and relative  translation. At each orientation during gyre refinement, the amplitudes (but not the phases) of the structure factors of the symmetry-related copies of the molecular-replacement fragments oriented (but not positioned) in the unit cell can be calculated, giving a set of normalized structure-factor amplitudes for the rotating components.</p>
<p>In gyre refinement, an initial RMSD parameter is chosen as a tradeoff between convergence radius and sensitivity to coordinate accuracy, iterating refinement and decreasing the RMSD parameter estimation sequentially. The goal is to improve and select among the many possible models those with a true r.m.s.d. of below 0.6 A˚ , and thus susceptible of being expanded to the full solution in the density-modification and autrotracing step. The chain definition also changes between cycles in order to increase the number of fragments and thus the degrees of freedom for model refinement, as predefined in the templatepartitioning step.</p>
<p>After model translation, packing filtering and refinement, models are either disassembled into predetermined rigid groups and refined (gimble refinement)</p>
</div>
<div class="section" id="perform-gimble-refinement">
<h3><span class="section-number">7.3.1.3. </span>Perform gimble refinement<a class="headerlink" href="#perform-gimble-refinement" title="Permalink to this headline">¶</a></h3>
<p>GIMBLE refinement is used for gimble the refinement of rigid-body fragments of the model after positioning. it is simply a re-implementation of Phaser’s rigid-body refinement developed for ease of scripting. It performs the refinement of rigid-body groups against the translation function target.</p>
</div>
<div class="section" id="perform-llg-guided-pruning">
<h3><span class="section-number">7.3.1.4. </span>Perform LLG-guided pruning<a class="headerlink" href="#perform-llg-guided-pruning" title="Permalink to this headline">¶</a></h3>
<p>LLG-guided pruning is used to trim the model of residues that are not contributing signal to the LLG at the target rmsd value.</p>
<p>To find out more please read article <a class="reference external" href="https://dx.doi.org/10.1107%2FS2059798318004357" rel="noopener noreferrer" target="_blank">here</a></p>
<blockquote>
<div>more Shredder tutorial you can find on <a class="reference external" href="http://chango.ibmb.csic.es/tutorial_shredder" rel="noopener noreferrer" target="_blank">here</a> and <a class="reference external" href="http://chango.ibmb.csic.es/tutorial_shredder_spherical" rel="noopener noreferrer" target="_blank">here</a></div></blockquote>
<p><strong>References</strong></p>
<p><a class="reference external" href="https://doi.org/10.1107/S0021889807021206" rel="noopener noreferrer" target="_blank">McCoy, A.J., Grosse-Kunstleve, R.W., Adams, P.D., Winn, M.D., Storoni, L.C., Read R.J. (2007) Phaser Crystallographic Software. J. Appl. Cryst. 40: 658-674</a></p>
<p><a class="reference external" href="https://doi.org/10.1038/nmeth.2644" rel="noopener noreferrer" target="_blank">Sammito, M. D., Millán, C., Rodríguez, D. D., de Ilarduya, I. M., Meindl, K., De Marino, I., Petrillo, G., Buey, R. M., de Pereda, J. M., Zeth, K., Sheldrick, G. M. &amp; Usón, I. (2013) Exploiting tertiary structure through local folds for crystallographic phasing. Nature Methods, 10, 1099–1101</a></p>
<p><a class="reference external" href="https://doi.org/10.1111/febs.12897" rel="noopener noreferrer" target="_blank">Sammito, M., Meindl,K., Ilarduya,I. M., Millán, C., Artola-Recolons, C., Hermoso, J. A., Usón, I. (2014) Structure solution with ARCIMBOLDO using fragments derived from distant homology models. FEBS J. 281, 4029–4045.</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S2059798318001353" rel="noopener noreferrer" target="_blank">McCoy, A. J., Oeffner, R. D., Millán, C., Sammito, M., Usón, I., Read, R. J. (2018) Gyre and gimble: a maximum-likelihood replacement for Patterson correlation refinement. Acta Cryst. D74, 279-289,</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S2059798317015121" rel="noopener noreferrer" target="_blank">Uson, I., Sheldrick, G.M. (2018) An introduction to experimental phasing of macromolecules illustrated by SHELX; new autotracing features. Acta Cryst. D74: 106-116</a></p>
<p><a class="reference external" href="https://doi.org/10.1107/S2059798318004357" rel="noopener noreferrer" target="_blank">Oeffner, R.D, Afonine, P.V, Millán, C, et al. (2018) On the application of the expected log-likelihood gain to decision making in molecular replacement. Acta Crystallogr D Struct Biol.;74(Pt 4): 245-255</a></p>
</div>
</div>
</div>


                        
                    </div>
                </div>
            </div>
        </div>
    </div>    


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.7',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/contentui.js"></script>
    <script type="text/javascript" src="_static/js/theme.js"></script>
  
    <div class="footer" role="contentinfo">
        <div class="container">
        Created using <a href="http://sphinx-doc.org/">Sphinx</a> 4.0.1.
        </div>
    </div>  

</body>
</html>