
<!DOCTYPE html>

<html>
  <head>

    <script async src="https://www.googletagmanager.com/gtag/js?id=G-FCVD2T7BGM"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'G-FCVD2T7BGM');
    </script>
    
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3. GerE: SAD and MR-SAD Phasing &#8212; CCP4 Cloud 1.7 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/haiku.css" />
    <link rel="stylesheet" type="text/css" href="_static/contentui.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/contentui.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="4. Using PISA in CCP4 Cloud" href="doc.demo.PISA_start.html" />
    <link rel="prev" title="2. GerE: Auto-EP with Crank-2" href="doc.demo.003EP.html" /> 
  </head><body>
      <div class="header" role="banner"><h1 class="heading"><a href="index.html">
          <span>CCP4 Cloud 1.7 documentation</span></a></h1>
        <h2 class="heading"><span>3. GerE: SAD and MR-SAD Phasing</span></h2>
      </div>
      <div class="topnav" role="navigation" aria-label="top navigation">
      
        <p>
        «&#160;&#160;<a href="doc.demo.003EP.html"><span class="section-number">2. </span>GerE: Auto-EP with Crank-2</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="doc.demo.PISA_start.html"><span class="section-number">4. </span>Using PISA in CCP4 Cloud</a>&#160;&#160;»
        </p>

      </div>
      <div class="content" role="main">
        
        
  <div class="section" id="gere-sad-and-mr-sad-phasing">
<h1><span class="section-number">3. </span>GerE: SAD and MR-SAD Phasing<a class="headerlink" href="#gere-sad-and-mr-sad-phasing" title="Permalink to this headline">¶</a></h1>
<p>The data are from the crystal structure of GerE, a transcription activator from Bacillus subtilis, which was solved by MAD phasing using the Se signal (V.M.A. Ducros, R.J. Lewis, C.S. Verma, E.J. Dodson, G. Leonard, J.P. Turkenburg, G.N. Murshudov, A.J. Wilkinson and J.A. Brannigan, J. Mol. Biol. (2001) 306 759-771).</p>
<p>The protein contains 70 amino acids including two (Se-)Met. Crystal structure has 3 dimers in the asymmetric unit. With modern software, the structure can also be easily solved by SAD phasing from just the Se peak or inflection data.</p>
<p>The paths to files used in this tutorial:</p>
<p>tutorial-data/2_phasing/gere_SAD/gere.mtz</p>
<p>tutorial-data/2_phasing/gere_SAD/gere_nat.mtz</p>
<p>tutorial-data/2_phasing/gere_SAD/gere.fasta</p>
<p>tutorial-data/2_phasing/gere_SAD/gere_molA.pdb</p>
<p>The above paths are relative to the cloud storage root.</p>
<p>Objectives</p>
<ol class="arabic simple">
<li>To learn how to operate with CCP4 Cloud web-interface.</li>
<li>To get familiar with individual steps of experimental phasing.</li>
</ol>
<p>This tutorial continues “GerE: Auto-EP with Crank-2” and does not repeat explanations of structure revision etc. contained there.</p>
<p>Workflow is fully defined by numbered paragraphs. Open cycles mark additional information.</p>
<p>Workflow</p>
<ol class="arabic simple">
<li>At this point you have already created a tutorial project and imported native and Se-peak diffraction data, and sequence.</li>
</ol>
<p>2. Create an initial revision suitable for experimental phasing. Define Se as anomalous scatterer.
Labels of data objects, which appear in menus, look similar to this one: “[0001-02] gere_nat [MADval/Gere/nat] /hkl/anom/”. The first item, “0001-002”, is the data object identifier formed by job number and the index of the data object within the job that created it. The second item, “gere_nat [MADval/Gere/nat]”, is the data-object descriptive name, which, for reflection data objects, consists of  mtz filename without extension and mtz-dataset name. The last item, “/hkl/anom/”, show type and subtype(s) of the data object. It may happen that an mtz-dataset does not actually have anomalous signal but contains mtz-columns I+ and I- (or F+ and F-) and, therefore, is assigned the subtype “anom”. Be careful, the reflection data object “gere_nat [MADval/Gere/nat]”  is such an example. It has subtype “anom” but does not actually have any anomalous signal.</p>
<p>3. Use “Substructure Search with SHELX via Crank-2 (SAD)” to find anomalous substructure.
How many Se sites has been found?
There is similar task “Substructure Search with SHELX-C/D (SAD)”, which does not generate graphical report page, but instead shows the original SHELXD output in the tab “Main Log”. Unfortunately, this task requires a fix for large intensities and wont currently work with GerE data.</p>
<p>4. Use “Experimental Phasing with Phaser (SAD)” for substructure completion and initial phasing.
“Se” under the “Structure revision” means that the input substructure contains Se (as should have been defined in the input of “Asymmetric Unit Contents” task). To extend the substructure with Se, you will have to define “Se” once again,  in the LLG map completion menu.
How many Se sites are there after substructure completion?</p>
<ol class="arabic simple" start="5">
<li>The Phaser-SAD job generates two structure revisions, for original and inverse hands. Perform density modification for each of them using “Density Modification with Parrot”. Use corresponding substructure as “Model for NCS detection”.</li>
</ol>
<p>Parrot uses substructure to define initial values of NCS operations. However, each monomers contain only two Se atoms. This is insufficient for determination of NCS operations relating monomers. How Parrot managed to figure out the NCS operations?</p>
<p>Pay attention to “Reflection statistics” and “NCS statistics” plots, specifically to changes of FOM and NCS mask volume with cycle. Which hand is correct? Could more cycles of density modification be useful? You may run another Parrot job if you think they could.</p>
<p>The revision consists of several data objects. (Use “Inspect” button to see them). The current job can only see the revision(s) from the parent job. However, the model for NCS detection is not a revision but a data object of type “structure” or “substructure” (a constituent  part of a certain revision or a stand-alone data object). The current job can see relevant data objects from all the jobs on its branch. Therefore, the “Model for NCS detection” can be taken from earlier jobs, and, in particular, it can be a structure from an MR job.</p>
<p>6. Inspect electron density generated by the two Parrot runs (UglyMol viewer can be opened from Report tab of finished Parrot job window).
Does comparison of the two maps confirms the correct hand identified from stats?</p>
<ol class="arabic simple" start="7">
<li>Continue each of the Parrot jobs with “Automatic Model Building with Buccaneer”.</li>
</ol>
<p>The correct hand should now be identified with certainty.</p>
<p>In a real project you would continue the best Buccaneer job by iterating “Model Building with Coot” and “Refmac5” (see Refinement tutorial part 3).</p>
<p>The rest of this tutorial demonstrates MR-SAD.</p>
<ol class="arabic simple" start="8">
<li>We have already imported gere_molA.pdb as coordinate object. Use “Ensemble Preparation for MR from Coordinates” to convert this into ensemble object. Associate the ensemble with the imported gere sequence, but do not do any modification - this is already a perfect model of a single monomer.</li>
</ol>
<p>The ensemble search model is the model composed of several superposed homologues molecules. A single-chain model can be considered as a single-chain ensemble model. Such a terminology, where ensemble = search model was initially adopted in Phaser and then in other programs including ccp4cloud.</p>
<ol class="arabic simple" start="9">
<li>Run “Molecular Replacement with Molrep”. Ask for 3 copies - we want to emulate partial MR solution that requires additional phase information for model completion.</li>
</ol>
<p>Use UglyMol to inspect electron density map. Can you locate position of missing molecules? Is the density good enough for manual model building?</p>
<ol class="arabic simple" start="10">
<li>Use “Experimental Phasing with Phaser (SAD)” for Se-substructure determination and initial phasing.</li>
</ol>
<p>Compare this Phaser job with (4): FOM, number of sites. Is the substructure likely to be correct?</p>
<p>How many revision were generated by this Phaser-EP job in (4) and here? Why?</p>
<ol class="arabic simple" start="11">
<li>Continue as in (5-7).</li>
<li>Compare structures obtained on SAD and MR-SAD routes.</li>
</ol>
<p>12.1. Select the job that produced a structure that you want use as reference (Job A).</p>
<p>12.2. Press and keep Apple key (Mac) or Ctrl key (Linux) and select the job that produced the structure you want to match to the reference (Job B).</p>
<p>12.3 Open task “Symmetry Match to Reference Structure with CSymMatch”. Check that selected revision is the one produced by (Job B). In the reference structure drop-down menu, select the structure produced by (Job A).</p>
<p>Note that the CSymMatch job node shows its own job number and the number of the job where additional data came from.</p>
<p>12.4 Continue CSymMatch job with Coot. In Coot task interface, select the structure from (Job A) in the  “Additional structure” menu. Run Coot. The models from SAD and MR-SAD branches should be superposed and you will be able to see differences.</p>
</div>


      </div>
      <div class="bottomnav" role="navigation" aria-label="bottom navigation">
      
        <p>
        «&#160;&#160;<a href="doc.demo.003EP.html"><span class="section-number">2. </span>GerE: Auto-EP with Crank-2</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="doc.demo.PISA_start.html"><span class="section-number">4. </span>Using PISA in CCP4 Cloud</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer" role="contentinfo">
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.0.1.
    </div>
  </body>
</html>